### PR TITLE
Added pl-paired-vector

### PR DESCRIPTION
**Pull request by SanjitBasker**
Added new element pl-paired-vector, which is to be used for equal and opposite forces within the same pl-diagram. Grading is as follows:
- each of the 2 student arrows is compared to each of the 2 answer arrows
- if student arrows are matched to answer arrows, and the difference in angle between the student arrows is 180°±2*angtol, then full credit is given
- disregard-sense can be used to accept both possible orientations

Example questions using these reactions are in demo/drawing/inclinedPlane-reaction and demo/drawing/frame-exploded.
I can't load the svg file correctly (which I've saved as pl-paired-vector-custom.svg), so the current pl-paired-vector.svg is just a copy of pl-line.svg.

### CODE REVIEW

**Comment by SanjitBasker** [comment]
I have reviewed and hereby sign the CLA

File: elements/pl-drawing/mechanicsObjects.js [hunk 1]
```diff
@@ -2788,6 +2788,158 @@ mechanicsObjects.byType['pl-vector'] = class extends PLDrawingBaseElement {
...
+                canvas.remove(obj1);
+                canvas.remove(obj2);
+            });
+
+        mechanicsObjects.attachHandlersNoClone(subObj, obj2, submittedAnswer, 
+            function(){
+            for (const key of ['left', 'top', 'angle']) {
+                subObj[key.concat('2')] = obj2[key];
+                    console.log(key.concat('2'));
+                    console.log(subObj[key.concat('2')]);
```
Comment (nicknytko) [inline elements/pl-drawing/mechanicsObjects.js:2927]:
```suggestion
```

File: elements/pl-drawing/pl-drawing.js [hunk 2]
```diff
@@ -321,9 +321,9 @@ class PLDrawingAnswerState {
      * This maintains a "submission" object that is separate from the canvas object.
      * By default, all properties from the canvas object are copied to the submission object.
      *
-     * @options options Options that were passed to the 'generate()' function.
-     * @options object Canvas object that was created and should be saved.
-     * @modifyHandler {optional} Function that is run whenever the canvas object is modified.
+     * @param options Options that were passed to the 'generate()' function.
+     * @param object Canvas object that was created and should be saved.
+     * @param modifyHandler {optional} Function that is run whenever the canvas object is modified.
```
Comment (nicknytko) [inline elements/pl-drawing/pl-drawing.js:326]: nice catch on these!

File: elements/pl-drawing/pl-drawing.js [hunk 3]
```diff
@@ -361,9 +361,7 @@ class PLDrawingAnswerState {
         this.updateObject(submitted_object);
         object.on('modified', () => {
             if (modifyHandler) {
-                console.log(submitted_object);
                 modifyHandler(submitted_object, object);
-                console.log(submitted_object);
```
Comment (nicknytko) [inline elements/pl-drawing/pl-drawing.js:366]: 😁

File: elements/pl-drawing/mechanicsObjects.js [hunk 4]
```diff
@@ -2788,6 +2788,158 @@ mechanicsObjects.byType['pl-vector'] = class extends PLDrawingBaseElement {
...
+            canvas.add(textObj2);
+        }
+
+        var subObj= mechanicsObjects.cloneMechanicsObject('pl-paired-vector', options);
+        mechanicsObjects.attachHandlersNoClone(subObj, obj1, submittedAnswer, 
+            function(){
+            for (const key of ['left', 'top', 'angle']) {
+                subObj[key.concat('1')] = obj1[key];
+                console.log(key.concat('1'));
+                console.log(subObj[key.concat('1')]);
```
Comment (nicknytko) [inline elements/pl-drawing/mechanicsObjects.js:2914]:
```suggestion
```

File: elements/pl-drawing/elements.py [hunk 5]
```diff
@@ -541,6 +541,134 @@ def get_attributes():
...
+                    st2[attr] = st2[attr + str(j + 1)]
+                for attr in ['x', 'y']:
+                    ref2[attr + '1'] = ref2[attr + str(i + 2)]
+                    st2[attr + '1'] = st2[attr + str(j + 2)]
+                poss[i][j] = Vector.grade(ref2, st2, tol, angtol)
+                counter += 1
+        angdiff = abs(st2['angle1'] - st2['angle2'])
+        angdiff = abs(angdiff - 180)
+        # print(angdiff < 2*angtol)
+        # print(poss)
```
Comment (nicknytko) [inline elements/pl-drawing/elements.py:665]:
```suggestion
```

File: elements/pl-drawing/elements.py [hunk 6]
```diff
@@ -541,6 +541,134 @@ def get_attributes():
...
+        counter = 0
+        for i in range(2):
+            for j in range(2):
+                for attr in dup_attrs:
+                    ref2[attr] = ref2[attr + str(i + 1)]
+                    st2[attr] = st2[attr + str(j + 1)]
+                for attr in ['x', 'y']:
+                    ref2[attr + '1'] = ref2[attr + str(i + 2)]
+                    st2[attr + '1'] = st2[attr + str(j + 2)]
+                poss[i][j] = Vector.grade(ref2, st2, tol, angtol)
```
Comment (nicknytko) [inline elements/pl-drawing/elements.py:660]: Nice!

**Review by nicknytko — COMMENTED** [review]
Took a brief pass at the code, generally looks good.  I'll make sure to take a deeper pass this weekend and actually run this locally. (Please remind me if I forget!)

**Review by nicknytko — COMMENTED** [review]
Just tried this out; looks great! Just a few things and I think we should be ready to merge:

- Can you get the custom icon working? I actually just tried renaming the `custom.svg` to `pl-paired-vector.svg` and it worked locally on my end, perhaps I'm misunderstanding your comment?
- Could you add your demo questions (`demo/drawing/inclinedPlane-reaction`, `demo/drawing/frame-exploded`) to the drawing element example assessment in `exampleCourse/courseInstances/Sp15/assessments/hw05-demoDrawingExamples/infoAssessment.json`?  Also, perhaps writing a sentence or two in the question itself introducing the new element.  Something like "this question uses the `pl-paired-vector` element to link two vectors together in the FBD...".
- Add a relevant entry in `docs/pl-drawing/index.md`.  I think you should be able to copy+paste the `pl-vector` entry for use as a base?

Thanks again for your work on this PR!

**Comment by SanjitBasker** [comment]
Awesome, I've updated the docs and drawing assessment. The svg icon is working for me as well, now. Not sure what was going on before haha.

**Pull request merged**